### PR TITLE
Add response header configuration

### LINKED ISSUE #113: Add response header configuration (Location, ETag, custom Content-Type, etc.)
## Summary
Allow configuring response headers on a mocked response. There is currently no way to set headers like `Location`, `ETag`, `Cache-Control`, or a custom content-type on the returned `HttpResponseMessage` short of using the low-level `RespondsWith(HttpContent)` / `RespondsWith(Func<...>)` overloads.

## Motivation
Code under test frequently reads response headers (e.g. follows `Location` after a `201 Created`, checks `ETag`/`Cache-Control`, reads a pagination or rate-limit header). Users should not have to drop down to hand-building `HttpResponseMessage` for this.

## Proposed API (additive)
On `RequestMockResponseBuilder`:
```csharp
RequestMockResponseBuilder WithHeader(string name, string value);
RequestMockResponseBuilder WithHeader(string name, params string[] values);
```
Usage:
```csharp
mock.ForPost().WithPath("/api/users")
    .RespondsWithStatus(HttpStatusCode.Created)
    .WithHeader("Location", "/api/users/123");
```

### dennisdoomen/mockly (core)
- `RequestMockResponseBuilder` currently only holds a `RequestMock` and mutates `MaxInvocations`. Extend it to wrap the mock's `Responder` so that, after it produces the message, the configured headers are applied (response headers vs. content headers must be routed correctly — e.g. `Content-Type`/`Content-Length` go on `response.Content.Headers`, others on `response.Headers`).
- Apply headers using `TryAddWithoutValidation` to support non-standard headers.
- Edge case: responders that return a message with no `Content` when a content header is requested — create an empty `StringContent` or document the limitation.
- Tests: standard header, content header routing, multi-value header, header applied on every invocation.

### dennisdoomen/fluentassertions.mockly (assertions — **companion work**)
- `CapturedRequestAssertions` and `ContainedRequestAssertions` assert on *request* properties today. Since `CapturedRequest` holds the full `HttpResponseMessage`, add **response-header assertion** methods:
  ```csharp
  WithResponseHeader(string name, string because = "")
  WithResponseHeader(string name, string valuePattern, string because = "")
  ```
  These assert that the captured response included the expected header, closing the loop between configuring and verifying response headers.
- Both `v7`/`v8` packages must be updated via `Shared/`.
- Needs `AcceptApiChanges.ps1` in the FA.Mockly repo.

## Breaking-change analysis
- **None for consumers.** Additive methods on `RequestMockResponseBuilder`.
- Internally this changes how `Responder` is invoked (wrapping), but `RequestMock.Responder` remains a public `Func<RequestInfo, HttpResponseMessage>` and keeps working.
- Core: `api-approved` workflow + `AcceptApiChanges.ps1` in `dennisdoomen/mockly`.
- FA.Mockly: `AcceptApiChanges.ps1` in `dennisdoomen/fluentassertions.mockly`.

### PR DESCRIPTION
**Pull request by dennisdoomen**
Closes #113

## Summary

Adds the ability to configure response headers on a mocked response, so consumers no longer need to drop down to hand-building an `HttpResponseMessage` to set headers like `Location`, `ETag`, `Cache-Control`, or a custom `Content-Type`.

Two additive, fluent methods are added to `RequestMockResponseBuilder`:

```csharp
RequestMockResponseBuilder WithHeader(string name, string value);
RequestMockResponseBuilder WithHeader(string name, params string[] values);
```

Usage:

```csharp
mock.ForPost().WithPath("/api/users")
    .RespondsWithStatus(HttpStatusCode.Created)
    .WithHeader("Location", "/api/users/123")
    .WithHeader("ETag", "\"v1\"");
```

## How it works

- The mock's `Responder` is wrapped so that, after it produces the `HttpResponseMessage`, the configured headers are applied on **every** invocation.
- Content headers (`Content-Type`, `Content-Length`, `Content-Encoding`, `Content-Disposition`, `Expires`, `Last-Modified`, etc.) are routed to `response.Content.Headers`; all other headers go to `response.Headers`.
- Headers are added with `TryAddWithoutValidation`, so non-standard headers are supported.
- If a content header is configured but the response has no content, an empty `ByteArrayContent` is created to host it.
- The same header configured twice results in the last value winning (remove-then-add semantics).
- The header name is validated at configuration time (rejects null/empty/white-space) and the values array is defensively copied so later mutation of the caller's array doesn't affect responses.

## New public API

- `RequestMockResponseBuilder.WithHeader(string name, string value)`
- `RequestMockResponseBuilder.WithHeader(string name, params string[] values)`

The approved API verification files (`net472.verified.txt`, `net8.0.verified.txt`) were regenerated via `AcceptApiChanges.ps1`.

## Tests

Added a `ResponseHeaders` spec class in `Mockly.Specs` (xUnit, FluentAssertions, AAA) covering: standard header, content-header routing, empty-content creation, multi-value header, header applied on every invocation, chaining, last-value-wins, configuration-time name validation, and array-mutation safety. Full suite is green on `net472` and `net8.0`.

## Breaking changes

None. The methods are purely additive. `RequestMock.Responder` remains a public `Func<RequestInfo, HttpResponseMessage>`.

## Follow-up

This issue also describes companion response-header **assertions** for the separate `dennisdoomen/fluentassertions.mockly` repository (`WithResponseHeader` on `CapturedRequestAssertions`/`ContainedRequestAssertions`, for both the v7 and v8 packages). That work is **not** part of this repository and requires a separate PR in `dennisdoomen/fluentassertions.mockly`.

### CODE REVIEW

File: Mockly/RequestMockResponseBuilder.cs [hunk 1]
```diff
@@ -43,4 +62,75 @@
...
+        };
+
+        return this;
+    }
+
+    private static void ApplyHeader(HttpResponseMessage response, string name, string[] values)
+    {
+        if (ContentHeaderNames.Contains(name))
+        {
+            response.Content ??= new ByteArrayContent([]);
```
[the remark this answers is not in the document]
Comment (dennisdoomen) [reply Mockly/RequestMockResponseBuilder.cs:134]: Fixed in fb198e9. On net8.0 `HttpResponseMessage.Content` is non-null by default, so the `??=` left operand was indeed never null there. I've scoped the empty-content fallback to the net472 target (where `Content` does default to null), removing the redundant check from the net8.0 build.
